server/src/main/java/io/druid/server/ClientInfoResource.java

```diff
@@ -83,13 +100,83 @@ public ClientInfoResource(
   @Produces(MediaType.APPLICATION_JSON)
   public Map<String, Object> getDatasource(
```

Comment:
404 really means "url not found", which is not the case here. theoretically speaking, this is a "400 bad request" case.
but I will make it 404 to be consistent with coordinator behavior.